### PR TITLE
Add target components logic into `HierarchicalPipeline`

### LINKED ISSUE #1170: Add target components logic into `HierarchicalPipeline`
### 🚀 Feature Request

Add target components logic into `HierarchicalPipeline`

### Proposal

In `HierarchicalPipeline`
- Create methods `predict` and `raw_predict`, implement them the same way as `forecast`, `raw_forecast` but calling the `predict`
- In methods `forecast`, `predict` pass `return_components` flag to `raw_forecast`,`raw_predict` 
- In methods `raw_forecast`,`raw_predict`
    - Pass `return_components` flag to to the methods `forecast`, `predict` of the super class
    - Add target components into resulting dataset in the end of the methods using `add_target_components`

### Test cases

- Create tests for `predict`, `raw_predict` similar to `forecast`, `raw_forecast`
-  Check that when `return_components=True` dataset contains target components(`ts.target_components is not None`) for methods `forecast`, `predict`

### Additional context

Is blocked by #1167, #1169

### PR DESCRIPTION
**Pull request by brsnw250**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Closing issues

closes #1170

### CODE REVIEW

**Review by alex-hse-repository — CHANGES_REQUESTED** [review]

File: etna/pipeline/mixins.py [hunk 1]
```diff
@@ -36,7 +36,15 @@ def _create_ts(self, ts: TSDataset, start_timestamp: pd.Timestamp, end_timestamp
...
 
         df_to_transform = df[:end_timestamp]
-        cur_ts = TSDataset(df=df_to_transform, df_exog=df_exog, freq=freq, known_future=known_future)
+
+        cur_ts = TSDataset(
+            df=df_to_transform,
+            df_exog=df_exog,
+            freq=freq,
+            known_future=known_future,
+            hierarchical_structure=ts.hierarchical_structure,
```
[the remark this answers is not in the document]
Comment (brsnw250) [reply etna/pipeline/mixins.py:45]: We don't provide any possibility to change `HierarchicalStructure` after creation. So here we can safely use references. Also, in other similar cases, we use reference as well.

File: tests/test_pipeline/test_hierarchical_pipeline.py [hunk 2]
```diff
@@ -425,19 +545,55 @@ def test_forecast_given_ts_with_prediction_interval(
...
+@pytest.mark.parametrize(
+    "model_fixture",
+    (
+        "non_prediction_interval_context_ignorant_dummy_model",
+        "non_prediction_interval_context_required_dummy_model",
+        "prediction_interval_context_ignorant_dummy_model",
+        "prediction_interval_context_required_dummy_model",
+    ),
+)
+def test_forecast_with_return_components(
```
[the remark this answers is not in the document]
Comment (brsnw250) [reply tests/test_pipeline/test_hierarchical_pipeline.py:557]: Here everything works as expected. For `forecast` `DummyModelBase` returns 100 (10 + 90 for forecast components), it is forecast for source `product` level. Also, in this test we do a bottom up reconciliation to obtain values at the target `market` level, so we sum up different segments (X = a + b = 100 + 100 = 200). Same holds for Y segment and components logic.

File: tests/test_pipeline/test_hierarchical_pipeline.py [hunk 3]
```diff
@@ -425,19 +545,55 @@ def test_forecast_given_ts_with_prediction_interval(
...
+    ),
+)
+def test_predict_with_return_components(
+    product_level_constant_hierarchical_ts, model_fixture, request, expected_component_a=40, expected_component_b=360
+):
+    ts = product_level_constant_hierarchical_ts
+    model = request.getfixturevalue(model_fixture)
 
-@to_be_fixed(NotImplementedError, "Adding target components is not currently implemented!")
-def test_raw_forecast_with_return_components(product_level_constant_hierarchical_ts):
```
[the remark this answers is not in the document]
Comment (brsnw250) [reply tests/test_pipeline/test_hierarchical_pipeline.py:438]: Generally, they will test the same logic as `predict` and `forecast` in terms of obtaining components. Alternatively, we can test just `raw_predict`  and `raw_forecast`, as they don't perform reconciliation, which is tested separately.

File: etna/pipeline/hierarchical_pipeline.py [hunk 4]
```diff
@@ -5,6 +5,8 @@
 from typing import Optional
```
[the remark this answers is not in the document]
Comment (brsnw250) [reply etna/pipeline/hierarchical_pipeline.py:5]: Added

**Review by alex-hse-repository — APPROVED** [review]

**Pull request merged**